runtime/src/invoke_dpe.rs

```diff
@@ -145,28 +145,36 @@ impl InvokeDpeCmd {
                 Command::GetCertificateChain(cmd) => cmd.execute(dpe, &mut env, locality),
             };
 
+            let mut invoke_resp = InvokeDpeResp {
+                hdr: MailboxRespHeader::default(),
+                data_size: 0,
+                data: [0u8; InvokeDpeResp::DATA_MAX_SIZE],
+            };
+
             // If DPE command failed, populate header with error code, but
             // don't fail the mailbox command.
-            let resp_struct = match resp {
-                Ok(r) => r,
-                Err(e) => {
+            match resp {
+                Ok(ref r) => {
+                    let resp_bytes = r.as_bytes();
+                    let data_size = resp_bytes.len();
+                    invoke_resp.data[..data_size].copy_from_slice(resp_bytes);
+                    invoke_resp.data_size = data_size as u32;
+                }
+                Err(ref e) => {
                     // If there is extended error info, populate CPTRA_FW_EXTENDED_ERROR_INFO
                     if let Some(ext_err) = e.get_error_detail() {
                         drivers.soc_ifc.set_fw_extended_error(ext_err);
                     }
-                    Response::Error(ResponseHdr::new(e))
+                    let r = ResponseHdr::try_mut_from_bytes(
```

Comment:
Since `Response::Error::as_bytes` calls the inner `ResponseHdr::as_bytes` the data being serialized has not changed.